src/transport/MessageCounter.h

```diff
@@ -71,6 +72,12 @@ class GlobalUnencryptedMessageCounter : public MessageCounter
         ++value;
         return CHIP_NO_ERROR;
     }
+    CHIP_ERROR SetCounter(uint32_t count) override
```

Comment:
When would we ever want to set this value for this counter type?

Reply:
This is the counter used for local message counting (as opposed to the `PeerMessageCounter`). This counter has no mechanism to set its value to a specified count, so instead of reset+advance in a for loop, I added a SetCounter API to just reset+set it to the desired value. 

CHIPDevice is doing this when a session is established.

Reply:
Right, but my understanding is that in the spec this counter is meant to just be set to a random value on boot...  But given that we then do value enforcement for it, that seems like a pretty serious spec bug!  @turon @tcarmelveilleux how is this part supposed to work?  The random value can randomly fall into a range the other side will consider a duplicate!

For now, I guess we keep this; it's almost certain that _this_ is why we were seeing handshake failures randomly: it depended on where the counter value fell.

Reply:
I filed https://github.com/project-chip/connectedhomeip/issues/7108 to track part of this....

Reply:
Let's continue the discussion in #7108

Reply:
To be clear, this PR and the problem it's trying to solve is not related to #7108 per se.  This PR is about the fact that our code has a serialize/deserialize in the middle of what is conceptually a single PASE session from the spec's point of view.  And if we do that, we need to keep counters across that operation, since it's the same session.